File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaUserApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaUserApi {

    @POST("/api/v3/user/create-sub")
    Call<ResponseResult> createSub(@Body Map<String, Object> paramMap);

    @POST("/api/v3/user/modify-sub")
    Call<ResponseResult> modifySub(@Body Map<String, Object> paramMap);

    @POST("/api/v3/user/freeze-sub")
    Call<ResponseResult> freezeSub(@Body Map<String, Object> paramMap);

    @GET("/api/v3/user/sub-list")
    Call<ResponseResult> subList(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/user/create-sub-api")
    Call<ResponseResult> createSubApi(@Body Map<String, Object> paramMap);

    @POST("/api/v3/user/update-sub-api")
    Call<ResponseResult> updateSubApi(@Body Map<String, Object> paramMap);

    @POST("/api/v3/user/delete-sub-api")
    Call<ResponseResult> deleteSubApi(@Body Map<String, Object> paramMap);

    @POST("/api/v3/user/sub-account/agent-create")
    Call<ResponseResult> agentCreate(@Body Map<String, Object> paramMap);

    @GET("/api/v3/user/sub-api-list")
    Call<ResponseResult> subApiList(@QueryMap Map<String, String> paramMap);
}

File: bitget-node-sdk-api/src/lib/v3/UtaUserApi.ts
import {BaseApi} from '../BaseApi';

export class UtaUserApi extends BaseApi {

    // POST /api/v3/user/create-sub
    createSub(qsOrBody: object) {
        const url = '/api/v3/user/create-sub';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/user/modify-sub
    modifySub(qsOrBody: object) {
        const url = '/api/v3/user/modify-sub';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/user/freeze-sub
    freezeSub(qsOrBody: object) {
        const url = '/api/v3/user/freeze-sub';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/user/sub-list
    subList(qsOrBody: object) {
        const url = '/api/v3/user/sub-list';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/user/create-sub-api
    createSubApi(qsOrBody: object) {
        const url = '/api/v3/user/create-sub-api';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/user/update-sub-api
    updateSubApi(qsOrBody: object) {
        const url = '/api/v3/user/update-sub-api';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/user/delete-sub-api
    deleteSubApi(qsOrBody: object) {
        const url = '/api/v3/user/delete-sub-api';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/user/sub-account/agent-create
    agentCreate(qsOrBody: object) {
        const url = '/api/v3/user/sub-account/agent-create';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/user/sub-api-list
    subApiList(qsOrBody: object) {
        const url = '/api/v3/user/sub-api-list';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}
